tests/js/common/shell/shell-pregel-components.js

```diff
@@ -412,6 +425,326 @@ function wccRegressionTestSuite() {
   };
 }
 
+function wccTestSuite() {
+  'use strict';
+
+  // componentsSizes should be an array of distinct sizes for components
+  // makeComponent should be a function that gets (<number of vertices>, <vertex collection name>, <name_prefix>)
+  // and returns {vertices: <array of vertices>, edges: <array of edges>}
+  const testWCCDisjointComponents = function(componentSizes, makeComponent) {
+    // we expect that '_' appears in the string (should it not, return the whole string)
+    const extract_name_prefix = function(v) {
+      return v.substr(0, v.indexOf('_'));
+    };
+
+    // expected components: [2_0, 2_1], [10_0,..., 10_9], [5_0,...,5_4], [23_0,...,23_22]
+    // We need for the test later that all sizes are distinct.
+    // Let's check that we didn't make a mistake.
+    const sortedComponentSizes = componentSizes.sort(function(a, b) { return b - a;});
+    for (let i = 0; i < sortedComponentSizes.size - 1; ++i) {
+      assertNotEqual(sortedComponentSizes[i], sortedComponentSizes[i+1],
+          `Test error: component sizes should be pairwise distinct, instead got ${componentSizes}`);
+    }
+
+    // Produce the graph.
+    for (let i of sortedComponentSizes) {
+      let {vertices, edges} = makeComponent(i, vColl, i.toString());
+      db[vColl].save(vertices);
+      db[eColl].save(edges);
+    }
+
+    // Get the result of Pregel's WCC.
+    const status = pregelRunSmallInstance("wcc", graphName, { resultField: "result", store: true });
+    assertEqual(status.state, "done", "Pregel Job did never succeed.");
+
+    // Now test the result.
+    // We expect four components.
+    const query = `
+        FOR v IN ${vColl}
+          COLLECT component = v.result WITH COUNT INTO size
+          SORT size DESC
+          RETURN {component, size}
+      `;
+    const computedComponents = db._query(query).toArray();
+    // assertEqual(computedComponents.length, sortedComponentSizes.length,
+    //     `We expected ${sortedComponentSizes.length} components, instead got ${JSON.stringify(computedComponents)}`);
```

Comment:
Is this intentionally a comment?
Either activate this assertion, or remove the comment please

Reply:
Activated